dev/set_matrix.py

```diff
@@ -0,0 +1,487 @@
+
+
+def remove_comments(s):
+    """
+    Examples
+    --------
+    >>> code = '''
+    ... # comment 1
+    ...  # comment 2
+    ... echo foo
+    ... '''
+    >>> remove_comments(code)
+    'echo foo'
+    """
+    return "\n".join(l for l in s.strip().split("\n") if not l.strip().startswith("#"))
+
+
+def make_pip_install_command(packages):
+    """
+    Examples
+    --------
+    >>> make_pip_install_command(["foo", "bar"])
+    "pip install 'foo' 'bar'"
+    """
+    return "pip install " + " ".join("'{}'".format(x) for x in packages)
+
+
+def divider(title, length=None):
+    r"""
+    Examples
+    --------
+    >>> divider("1234", 20)
+    '\n======= 1234 ======='
+    """
+    length = shutil.get_terminal_size(fallback=(80, 24))[0] if length is None else length
+    rest = length - len(title) - 2
+    left = rest // 2 if rest % 2 else (rest + 1) // 2
+    return "\n{} {} {}".format("=" * left, title, "=" * (rest - left))
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Set a test matrix for the cross version tests")
+    parser.add_argument(
+        "--ref-versions-yaml",
+        required=False,
+        default=None,
+        help=(
+            "URL or local file path of the reference config which will be compared with the config "
+            "on the branch where this script is running in order to identify version YAML updates"
+        ),
+    )
+    parser.add_argument(
+        "--changed-files",
+        type=lambda x: [] if x.strip() == "" else x.strip().split("\n"),
+        required=False,
+        default=None,
+        help=("A string that represents a list of changed files"),
+    )
+
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+
+    print(divider("Parameters"))
+    print(json.dumps(vars(args), indent=2))
+
+    ref_versions_yaml = (
+        VERSIONS_YAML_PATH if (args.ref_versions_yaml is None) else args.ref_versions_yaml
+    )
+    changed_files = [] if (args.changed_files is None) else args.changed_files
+
+    print(divider("Log"))
+    config = read_yaml(VERSIONS_YAML_PATH)
+    config_ref = read_yaml(ref_versions_yaml, if_error={})
+
+    # assuming that the top-level keys in `ml-package-versions.yml` have the format:
+    # <flavor name>(-<suffix>) (e.g. sklearn, tensorflow-1.x, keras-tf1.x)
+    flavors = set(x.split("-")[0] for x in config.keys())
+    changed_flavors = get_changed_flavors(changed_files, flavors)
+
+    # If both `--ref-versions-yaml` and `--changed-files` are unspecified, include all items.
+    # Otherwise, include only updated items by inspecting `config`, `config_ref`, and
+    # `changed_flavors`.
+    should_include_all_items = (args.ref_versions_yaml is None) and (args.changed_files is None)
+
+    job_names = []
+    includes = []
+
+    for flavor, cfgs in config.items():
+        package_info = cfgs.pop("package_info")
+
+        should_include_all_items_in_this_flavor = (
```

Comment:
Can we add a comment describing this boolean variable? It seems like it determines whether or not we need to run the full matrix of tests for the flavor or just the latest version, right?